Documentation: Add RFCs process document (#15687)
WordPress/gutenberg: docs/rfcs/README.md

# Requests for Comments (RFCs)

> To maintain an effective and transparent process for resolving complex technical problems, this project has implemented a Request for Comments (RFC) workflow.

Reviewer: Lead with what we are defining, not the why

```suggestion
The Gutenberg project uses a Request for Comments (RFC) workflow to maintain an effective and transparent process for resolving complex technical problems. 
```

For more information, refer to the blog post:

https://make.wordpress.org/core/2019/04/01/the-block-registration-rfc-and-the-gutenberg-rfc-process/

> This process is both inspired and informed by other open source projects such as [React](https://github.com/reactjs/rfcs), [Yarn](https://github.com/yarnpkg/rfcs), and [Rust](https://github.com/rust-lang/rfcs), and of technical bodies standardization processes such as [IETF](https://tools.ietf.org/html/rfc2026), [W3C](https://www.w3.org/2004/02/Process-20040205/tr.html), and [TC39](https://tc39.github.io/process-document/).

Reviewer: Minor edit for clarification

```suggestion
The Gutenberg RFC process is inspired and informed by other open source projects such as [React](https://github.com/reactjs/rfcs), [Yarn](https://github.com/yarnpkg/rfcs), and [Rust](https://github.com/rust-lang/rfcs), and of technical bodies standardization processes such as [IETF](https://tools.ietf.org/html/rfc2026), [W3C](https://www.w3.org/2004/02/Process-20040205/tr.html), and [TC39](https://tc39.github.io/process-document/).
```

- Submit a pull request to the Gutenberg repository with the Markdown file included in the `docs/rfc` folder ([see examples](/docs/rfcs/)).
- Discussion and feedback should take place in the pull request amongst the original author, Gutenberg Core team members, focus leads, project contributors, and any applicable [Core component maintainers](https://make.wordpress.org/core/components/). As the author of the proposal, you are encouraged to solicit feedback publicly, including to share your proposal in [Slack channels](https://make.wordpress.org/chat/) of interested teams.
- As a result of discussion in the pull request and at least one [weekly editor meeting](https://make.wordpress.org/meetings/), the RFC will be accepted (merged) or rejected (closed).

A merged RFC should serve as a sufficient baseline for an initial implementation, and as the de-facto documentation for the feature.

### RFC status

An RFC may be merged as a working document. The status of a document should be noted as a [YAML front matter (metadata)](https://github.blog/2013-09-27-viewing-yaml-metadata-in-your-documents/) in the first lines of the file, as one of the following:

#### `status: Draft`

> A "Draft" document is one which has been merged in a state sufficient for initial implementations, but which is expected to need future iterations to refine details or revise specifics which would only become apparent in the course of said implementations. The purpose of allowing for draft documents is to allow for a steady momentum of a feature proposal and to embrace quality via experimentation.

Reviewer: Tighten up a little, for example "refine details" and "revise specifics" are saying the same thing

```suggestion
A merged "Draft" document is sufficient for initial implementation, but expected to need future revisions based on details discovered during implementation. The purpose of allowing for draft documents is to allow for a steady momentum of a feature proposal and to embrace quality via experimentation.
```

Any document with a "Draft" status should be added as a column to the [Request for Comments (RFC) project](https://github.com/WordPress/gutenberg/projects/26) to serve as a reference for those interested to follow along with its progress.

A "Draft" document should always be kept in sync with the current direction of implementations using it as a base. Proposed changes to the RFC document should be assigned to the Project column.

#### `status: Standard`

Once an RFC document is considered to be finalized, its status should be migrated to "Standard".

> A document is final once its feature implementation is complete, and upon approval by focus leads.

Reviewer: ```suggestion
A document is final once its feature implementation is complete, and approved by focus leads.
```
